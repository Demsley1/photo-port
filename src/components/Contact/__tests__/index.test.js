import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import ContactForm from '..';

afterEach(cleanup);

describe('ContactForm Renders', () => {
    it('renders', () => {
        render(<ContactForm />)
    });

    it('matches with snapshot', () => {
        const{ asFragment } = render(<ContactForm />);
        expect(asFragment()).toMatchSnapshot();
    });
});

describe('Renders text content for title', () => {
    it('renders with h1 title', () => {
        const { getByTestId } = render(<ContactForm />);
        expect(getByTestId('contact-tag')).toHaveTextContent('Contact me')
        expect(getByTestId('contact-submit')).toHaveTextContent('Submit')
    })
})