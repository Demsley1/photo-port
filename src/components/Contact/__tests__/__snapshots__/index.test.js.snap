// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactForm Renders matches with snapshot 1`] = `
<DocumentFragment>
  <section>
    <h1
      data-testid="contact-tag"
    >
      Contact me
    </h1>
    <form
      id="contact-form"
    >
      <div>
        <label
          for="name"
        >
          Name:
        </label>
        <input
          name="name"
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          for="email"
        >
          Email address:
        </label>
        <input
          name="email"
          type="email"
          value=""
        />
      </div>
      <div>
        <label
          for="message"
        >
          Message:
        </label>
        <textarea
          name="message"
          rows="5"
        />
      </div>
      <button
        data-testid="contact-submit"
        type="submit"
      >
        Submit
      </button>
    </form>
  </section>
</DocumentFragment>
`;
